File: leetcode/153_findminimuminrotatedsortedarray.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        hi = len(nums) - 1
        lo = 0
        
        while lo < hi:
            mid = lo + (hi-lo) / 2
            if nums[mid] < nums[hi]:
                hi = mid
            else:
                lo = mid + 1
        return nums[lo]

File: leetcode/153_findminimuminrotatedsortedarray.ts
function findMin(nums: number[]): number {
  let l = 0
  let r = nums.length - 1

  while (l < r) {
    const mid = Math.floor((r + l) / 2)
    if (nums[mid] > nums[r]) { // min is on the right side
      l = mid + 1
    } else {
      r = mid
    }
  }
  
  return nums[r]
}; 
